### DISCUSSION TITLE
from gpt_engineer.db import DB, DBs, archive

### DISCUSSION THREAD

**Discussion by omar-ayoub** [AntonOsika/gpt-engineer#504] [Q&A]
ImportError: cannot import name 'archive' from 'gpt_engineer.db' 
does any one knows  why i get this error ?

**Comment by aadeelyounas** [comment]
I am also facing same error

**Comment by 3090goBRRR** [comment]
https://youtu.be/ceMuK0xUtSY?t=243

The above GPT-Engineer install video lead me here. Same error:

(gpt) C:\Users\XXXX\Desktop\gpt-engineer\gpt_engineer>PYTHON MAIN.PY EXAMPLE
Traceback (most recent call last):
  File "C:\Users\XXXX\Desktop\gpt-engineer\gpt_engineer\MAIN.PY", line 10, in <module>
    from gpt_engineer.db import DB, DBs, archive
ImportError: cannot import name 'archive' from 'gpt_engineer.db' (C:\Users\faked\.conda\envs\gpt\Lib\site-packages\gpt_engineer\db.py)

I notice that my install does not have the Example folder in the same location. Cannot properly execute main.py from any dir.

(Edit: Total re-install of GPT-Engineer did not resolve the issue.)

**Comment by jontstaz** [comment]
Hi,

Make sure you've got the latest commits (run 'git pull'). Then open the file `gpt_engineer/db.py` and check that the `archive` function exists. Assuming it does, reinstall gpt-engineer from source by running `pip install -e .` 
You shouldn't face that error as long as the archive function is in db.py. It was added only very recently so you may not have the most up-to-date files.

**Reply by cannacak** [reply]
I tried all the ways but still I have the same error.

**Reply by cannacak** [reply]
I sorted it out. **Use** 'gpt-engineer _PATH_' **not** 'python main.py _PATH_'

**Reply by AntiMatterZX** [reply]
it says cant export key

**Reply by joshscott95** [reply]
@cannacak right on this solved it.
however unless you have gpt 4 API key this just gives. 
```
INFO:openai:error_code=model_not_found error_message="The model 'gpt-4' does not exist" error_param=model error_type=invalid_request_error message='OpenAI API error received' stream_error=False
Model gpt-4 not available for provided API key. Reverting to gpt-3.5-turbo. Sign up for the GPT-4 wait list here: https://openai.com/waitlist/gpt-4-api


``` 

and then immediately after will throw this error

```
INFO:openai:error_code=insufficient_quota error_message='You exceeded your current quota, please check your plan and billing details.' error_param=None error_type=insufficient_quota message='OpenAI API error received' stream_error=False
WARNING:langchain.llms.base:Retrying langchain.chat_models.openai.ChatOpenAI.completion_with_retry.<locals>._completion_with_retry in 4.0 seconds as it raised RateLimitError: You exceeded your current quota, please check your plan and billing details..
WARNING:langchain.callbacks.manager:Error in StreamingStdOutCallbackHandler.on_retry callback: 'StreamingStdOutCallbackHandler' object has no attribute 'on_retry'

```

**Comment by Hockney87** [comment]
Same error here , when i select the 3.10 version of python the functions archive and collect_consent show that they can be imported .

**Comment by s4sourcing** [comment]
Hi.. Has anyone been able to solve this issue ?
